fraqbot/Local/shakespeare.py

```diff
@@ -0,0 +1,49 @@
+from Legobot.Lego import Lego
+import logging
+import os
+import random
+import sys
+
+logger = logging.getLogger(__name__)
+
+insult_array = []
+
+LOCAL_DIR = os.path.join(os.path.abspath(os.path.dirname(__file__)))
+if LOCAL_DIR not in sys.path:
+    sys.path.append(LOCAL_DIR)
+
+with open(os.path.join(LOCAL_DIR, 'lists/quotes.txt')) as my_file:
+    for line in my_file:
+        insult_array.append(line)
+
+
+class Shakespeare(Lego):
+    def __init__(self, baseplate, lock, *args, **kwargs):
+        super().__init__(baseplate, lock, acl=kwargs.get('acl'))
+
+    def listening_for(self, message):
+        return message['text'].split()[0] == '!shake'
```

Comment:
This can throw an Index Error if message['text'] is an empty string. It's also not guaranteed that it will be a string at all. Can we make it safer?
Maybe something like this?
```
text = message.get('text')
return isinstance(text, str) and text.startswith('!shake')
```